dataflow-website/recipes/scaling/scdf-alert-webhook/src/main/java/io/spring/cloud/dataflow/alert/webhook/AlertWebHookApplication.java

```diff
@@ -0,0 +1,141 @@
+ * stream application.
+ *
+ * The {@link AlertWebHookApplication} listens for scale-out and scale-in alerts, configured by the
+ * {@link AlertWebHookProperties#getScaleOutAlertName} and the {@link AlertWebHookProperties#getScaleInAlertName}.
+ *
+ * On scale-out alert it increases the number of application instances to {@link AlertWebHookProperties#getScaleOutFactor} instances.
+ *
+ * On scale-in alert it decreases the number of application instances to {@link AlertWebHookProperties#getScaleInFactor} instances.
+ *
+ * The stream name should be provided either as a {@link AlertWebHookProperties#getScaleStreamName} property or an
+ * Alert label called: 'stream_name'
+ *
+ * The application name should be provided either as a {@link AlertWebHookProperties#getScaleApplicationName} property or an
+ * Alert label called: 'application_name'
+ *
+ * Internally the {@link AlertWebHookApplication} uses the Data Flow Scale REST API to scale the app instances in platform agnostic way.
+ * Use the spring.cloud.dataflow.client.server-uri property to configure url of the Data Flow server.
+ *
+ * @author Christian Tzolov
+ */
+
+@SpringBootApplication
+@RestController
+@EnableConfigurationProperties(AlertWebHookProperties.class)
+public class AlertWebHookApplication {
+
+	private static final Logger logger = LoggerFactory.getLogger(AlertWebHookApplication.class);
+
+	public static final String FIRING_ALERT = "firing";
+	public static final String ALERT_NAME = "alertname";
+	public static final String ALERT_LABELS = "labels";
+	public static final String ALERT_STATUS = "status";
+	public static final String ALERTS = "alerts";
+	public static final String STREAM_NAME = "stream_name";
+	public static final String APPLICATION_NAME = "application_name";
+
+	private AlertWebHookProperties properties;
+
+	// Use the spring.cloud.dataflow.client.server-uri property to configure url of the Data Flow server.
+	private final DataFlowOperations dataFlowOperations;
+
+	@Autowired
+	public AlertWebHookApplication(DataFlowOperations dataFlowOperations, AlertWebHookProperties properties) {
+		this.properties = properties;
+		this.dataFlowOperations = dataFlowOperations;
+	}
+
+	/**
+	 * This method is called by the AlertManager passing a JSON payload as documented here:
+	 * https://prometheus.io/docs/alerting/configuration/#webhook_config
+	 */
+	@RequestMapping(value = "/alert", method = RequestMethod.POST)
+	public void alertMessage(@RequestBody Map<String, Object> payload) {
+
+		// Extract all alerts received with this message.
+		List<Map<String, Object>> alerts = (List<Map<String, Object>>) payload.get(ALERTS);
+
+		if (CollectionUtils.isEmpty(alerts)) {
+			return; // nothing to do
+		}
+
+		for (Map<String, Object> alert : alerts) {
+			String alertStatus = (String) alert.get(ALERT_STATUS);
+
+			// Act only upon "firing" alert types
+			if (alertStatus.equalsIgnoreCase(FIRING_ALERT)) {
```

Comment:
Maybe log the non-firing alerts in an else condition? Just so that we could see what other alert types are landing here.

Reply:
Makes sense